WIP for #55

groovecoder opened the pull request:
Latest UI updates to match [Panels Spec](https://www.dropbox.com/sh/jczak1kfm80q334/AAAvoNwz5lrcPZn1rreRD1E_a)

![testpilot-containers-ui-updates](https://cloud.githubusercontent.com/assets/71928/21899006/4f9d1fd8-d8b4-11e6-9068-03952b434302.gif)

groovecoder commented:
@bakulf - I updated this branch with a rebase no your latest merge to master, but I haven't finished cleaning up the container-info UI with your code. I'll be working on it Thursday, but wanted to put this here so you can see it before I wake up. 😉

groovecoder commented:
Okay, there are some big-ish changes here that I'd like to merge to master before we change more, and before I leave on PTO.

jonathanKingston commented on .gitignore:
```diff
@@ -3,3 +3,4 @@ node_modules
 README.html
 *.xpi
 *.swp
+.vimrc
```
👍

jonathanKingston commented on webextension/css/popup.css:
```diff
@@ -58,22 +89,49 @@ table {
 .edit-containers-text {
   text-align: center;
-  padding: .5em;
+  border-right: solid 1px #d8d8d8;
```
Can we make this `border-inline-end` ?

jonathanKingston commented on webextension/css/popup.css:
```diff
@@ -140,6 +198,40 @@ table {
+  block-size: 33px;
+  display: flex;
+  align-items: center;
+  border-bottom: 1px solid #ebebeb;
```
`border-block-end` and `margin-block-end`

jonathanKingston commented on webextension/css/popup.css:
```diff
@@ -140,6 +198,40 @@ table {
+}
+
+.container-info-panel-hideorshow-icon {
+  margin-right: 7px;
```
`margin-inline-end`

jonathanKingston commented on webextension/css/popup.css:
```diff
@@ -140,6 +198,40 @@ table {
+}
+
+.container-info-panel-movetabs {
+  border-bottom: 1px solid #ebebeb;
```
`border-block-end` and `margin-block-end`

jonathanKingston commented on webextension/img/container-newtab.svg:
```diff
@@ -0,0 +1,20 @@
+	.st0{fill:#979797;}
+</style>
+<title>icon-newtab</title>
+<desc>Created with Sketch.</desc>
```
Nit clean this out?

jonathanKingston commented on webextension/js/popup.js:
```diff
@@ -2,6 +2,13 @@
 const CONTAINER_UNHIDE_SRC = "/img/container-unhide.svg";
 
+function showPanel(panelSelector) {
+  for (let panelElement of document.querySelectorAll(".panel")) {
```
Are we doing const for these?

groovecoder replied:
We technically can, but since the statements in the loop modify a property of the object, I agreed with @bakulf that we should use `let` here to indicate that the variable does indeed change.

jonathanKingston replied:
Agreed, just wanted to check :).

jonathanKingston commented on webextension/js/popup.js:
```diff
@@ -16,6 +23,9 @@ function showContainerTabsPanel(identity) {
     trHasTabs.setAttribute("data-user-context-id", identity.userContextId);
   }
 
+  let hideOrShowRow = document.querySelector("#container-info-hideorshow");
```
const?

jonathanKingston commented on webextension/js/popup.js:
```diff
@@ -64,38 +76,33 @@ function showContainerTabsPanel(identity) {
-    element.classList.add("hide");
-  }
-  document.querySelector("#container-panel").classList.remove("hide");
+  showPanel("#container-panel");
```
👍

jonathanKingston commented on webextension/js/popup.js:
```diff
@@ -64,38 +76,33 @@ function showContainerTabsPanel(identity) {
-  document.querySelector(".onboarding-panel-2").classList.remove("hide");
-  document.querySelector(".onboarding-panel-1").classList.add("hide");
-  document.querySelector("#container-panel").classList.add("hide");
+  showPanel(".onboarding-panel-2");
```
This could be further simplified by calling a `checkCurrentStatus` method which could have the code above allowing the above code to call the same code also:
```
 if (localStorage.getItem("onboarded2")) {
```

jonathanKingston replied:
*not urgent/important either, just an observation.

jonathanKingston commented on webextension/js/popup.js:
```diff
@@ -129,13 +136,74 @@ browser.runtime.sendMessage({method: "queryIdentities"}).then(identities => {
+}
+
+document.querySelector("#add-container-link").addEventListener("click", () => {
+  showPanel("#edit-container-panel");
```
Should this call `showEditContainersPanel`?

groovecoder replied:
The `#edit-container-panel` is for editing a **single** container. The `#edit-containers-panel` (i.e., plural) is for editing the *list* of containers.

groovecoder commented:
Alright, think I addressed all the right comments. r? and merge?

groovecoder commented:
Okay, I added the color icons to the `edit-container-panel` and added a skeleton `delete-container-panel`.

I added it to this pull request in case it's the only thing I can get "done" today before going on PTO.

bakulf commented on webextension/js/popup.js:
```diff
@@ -32,12 +42,14 @@ function showContainerTabsPanel(identity) {
     method: "getTabs",
     userContextId: identity.userContextId,
   }).then(tabs => {
+    console.log('browser.runtime.sendMessage getTabs, tabs: ', tabs);
```
Let's remove this.

bakulf commented on webextension/js/popup.js:
```diff
@@ -64,38 +76,33 @@ function showContainerTabsPanel(identity) {
 });
 
 browser.runtime.sendMessage({method: "queryIdentities"}).then(identities => {
+  console.log('queryIdentities');
```
Instead, let's implement a debug() method and enable/disable it by pref.
